cmake/iotjs.cmake

```diff
@@ -25,27 +25,21 @@ function(find_value RESULT VALUE VALUE_TRUE VALUE_FALSE)
   endif()
 endfunction(find_value)
 
-# System Configuration
+# System Configuration (not module)
 set(IOTJS_PLATFORM_SRC)
-if("${CMAKE_SYSTEM_NAME}" MATCHES "Linux|Tizen")
-  file(GLOB IOTJS_PLATFORM_LINUX_SRC
-       ${IOTJS_SOURCE_DIR}/platform/iotjs_*-linux.c)
-  list(APPEND IOTJS_PLATFORM_SRC ${IOTJS_PLATFORM_LINUX_SRC})
-  # in case of Tizen force the platform to Linux
-  string(REGEX REPLACE "tizen$" "linux"
-         PLATFORM_DESCRIPTOR "${PLATFORM_DESCRIPTOR}")
-endif()
+string(TOLOWER ${CMAKE_SYSTEM_NAME} IOTJS_SYSTEM_OS)
+set(PLATFORM_OS_SRC
+    ${IOTJS_SOURCE_DIR}/platform/${IOTJS_SYSTEM_OS})
+set(OS_SRC ${PLATFORM_OS_SRC}/iotjs_[^module]*${IOTJS_SYSTEM_OS}.c)
+file(GLOB IOTJS_OS_SRC ${OS_SRC})
+list(APPEND IOTJS_PLATFORM_SRC ${IOTJS_OS_SRC})
```

Comment:
Is it really necessary to do the globbing in so many steps? Wouldn't
```cmake
file(GLOB IOTJS_PLATFORM_SRC ${IOTJS_PLATFORM_DIR}/iotjs_[^module]*-${IOTJS_SYSTEM_OS}.c)
```
also achieve the same? (The example already applies some of the suggested changes.)